src/HOC/withLoader.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+import LoadingSpinner from '../components/LoadingSpinner/LoadingSpinner';
+
+const withLoader = (WrappedComponent) => {
```

Comment:
fyi 
HOC are considered to be not the best practice anymore:
https://devcore.io/en/react/hocs-vs-hooks-what-to-use-and-why/